### PR TITLE
feat(telemetry): enable telemetry by default

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Enables telemetry collection when an application is run with `ddtrace-run`.

## Checklist
- [x] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [x] Add additional sections for `feat` and `fix` pull requests.
- [x] Ensure tests are passing for affected code.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Motivation

Collect tracer configurations.

## Design 

Starts a telemetry writer which sends a payload of tracer configuration to the datadog agent. As of now one event is sent on application startup and another is sent when an application terminates.

## Testing strategy

Ensure telemetry test suite passes in ci.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: ddtrace/internal/telemetry/writer.py [hunk 1]
```diff
@@ -115,14 +115,14 @@ def periodic(self):
             try:
                 resp = self._send_request(telemetry_request)
                 if resp.status >= 300:
-                    log.warning(
+                    log.debug(
```
Comment (brettlangdon) [inline ddtrace/internal/telemetry/writer.py:118]: this should be it's own PR

**Comment by mabdinur** [comment]
Blocked by: https://github.com/DataDog/dd-trace-py/pull/3930

**Review by brettlangdon — COMMENTED** [review]
lgtm

**Comment by mabdinur** [comment]
@Mergifyio backport 1.3

**Review by P403n1x87 — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]
